### PR TITLE
Legacy build_dbcsr cleanup

### PR DESCRIPTION
**Pull request by hfp**
* Consolidated Makefile rules and share between CUDA/HIP (differences are in flags rather than rules).
* Ensure DBCSR is finalized after other offload libraries (upcoming OpenCL Offload reuses DBCSR).

### CODE REVIEW

**Comment by alazzaro** [comment]
I've triggered the HIP build (I had the suspect that your changes will conflict with the HIP way to build, and indeed...).

**Comment by hfp** [comment]
I believe, I can sustain the cleanup. I so far, I fixed the CUDA Pascal Regtest (and it was a one-liner). Main reason to come up with this PR is `calculate_norms.cpp`. Though, DBCSR as currently bolted into CP2K does not have it (yet). This PR is mostly about "fix building the upcoming DBCSR". The thing is, `calculate_norms.cpp` contains CUDA code but is not called `.cu` which breaks the entire/convoluted Makefile...

**Comment by alazzaro** [comment]
Check my PR then, it is fixed there... We can make a release and merge my PR if it s priority...

**Comment by alazzaro** [comment]
https://github.com/cp2k/cp2k/pull/3244

**Comment by hfp** [comment]
I did not notice the draft-PR, sorry. Hmm, it now also relies on `OFFLOAD_CC` (and `OFFLOAD_FLAGS`) at least for HIP. Your PR helped fixing the HIP-part of this PR (I tested upfront all the Docker images for CUDA, HIP and ROCm). I will give my PR one more try (as it still reduces the LOC in the Makefile), i.e., worst-case your Draft is obsolete or can brush-up more things if still necessary.

**Comment by alazzaro** [comment]
hold on, I'm doing a dbcsr RC and merge my PR first...

**Comment by alazzaro** [comment]
DBCSR 2.7.0-rc1 is out https://github.com/cp2k/dbcsr/releases/tag/v2.7.0-rc1. 
Waiting CP2K tests to finish before merging...

**Comment by hfp** [comment]
> hold on, I'm doing a dbcsr RC and merge my PR first...

ACK

**Comment by hfp** [comment]
The revised Makefile seems to at least catch more issues, e.g., `acc_dev.cpp:23:18: error: unused variable 'verbose_print' [-Werror,-Wunused-const-variable]` which present since a long time but never caught (this fails ROCm build by the time of writing).

**Comment by alazzaro** [comment]
Please, rebase (and sorry for the problem). I've included the new DBCSR which will give you the possibility to continue with OpenCL porting in CP2K.

File: exts/build_dbcsr/Makefile [hunk 1]
```diff
@@ -51,6 +51,7 @@ FYPPEXE      := $(TOOLSDIR)/build_utils/fypp/bin/fypp
 SRCDIR       := $(DBCSRHOME)/src
 TESTSDIR     := $(DBCSRHOME)/tests
 ARCHFILE     :=
+CXXSTD       := --std=c++14
```
Comment (alazzaro) [inline exts/build_dbcsr/Makefile]: is it `--` or `-`?

Comment (hfp) [reply exts/build_dbcsr/Makefile]: Good catch!

**Comment by hfp** [comment]
I consider [CUDA Pascal Regtest](https://storage.googleapis.com/cp2k-ci/run-cp2k-cuda-pascal-7c8d833c_report.txt) to be OK, apparently it is the `QS/regtest-tddfpt-force-2/h2o_f13.inp` fluctuating.

**Comment by alazzaro** [comment]
LGTM

**Pull request merged**